### ISSUE TITLE
apply own masks when training

### ISSUE THREAD

**Issue by 123liluky** [JiahuiYu/generative_inpainting#131]
Hi, I have 2346 masks the same size as train images like below and I want apply them to my train images instead of randomly producing masks.
![1536802314 1](https://user-images.githubusercontent.com/20832778/45461943-3cbef800-b738-11e8-8b15-75075ffced61.png)
So I changed codes in inpaint_ops.py:
def random_bbox(config):
       img_shape = config.IMG_SHAPES
       img_height = img_shape[0]
       img_width = img_shape[1]
       h = tf.constant(img_height)
       w = tf.constant(img_width)
       return (tf.constant(0), tf.constant(0), h, w)
def bbox2mask(bbox, config, name='mask'):
       def npmask(bbox, height, width, delta_h, delta_w):
           mask = np.zeros((1, height, width, 1), np.float32)
           ind = np.random.randint(2346)
           mask_pre = Image.open(my_mask_dir+str(ind)+".jpg").point(lambda i : 0 if i==0 else 1)
           mask_pre = np.array(mask_pre)
           mask[0,:,:,0] = mask_pre.astype(np.float32)
           return mask
       with tf.variable_scope(name), tf.device('/cpu:0'):
           img_shape = config.IMG_SHAPES
           height = img_shape[0]
           width = img_shape[1]
           mask = tf.py_func(
               npmask,
               [bbox, height, width,
                config.MAX_DELTA_HEIGHT, config.MAX_DELTA_WIDTH],
               tf.float32, stateful=False)
           mask.set_shape([1] + [height, width] + [1])
       return mask
But the inpainting results are strange because they are colored instead of gay and I saw from tensorboard that mask apply to training images was always the same one from my_mask_dir instead of randomly choosing from my_mask_dir. Could you help me? Thank you in advance.
![1536802601 1](https://user-images.githubusercontent.com/20832778/45462115-02a22600-b739-11e8-949f-55efe8823801.png)
My inpait.yml is:
##parameters
DATASET: 'celebahq'
RANDOM_CROP: False
VAL: False
LOG_DIR: 'logs'
MODEL_RESTORE: ''  #

GAN: 'wgan_gp'  # 'dcgan', 'lsgan', 'wgan_gp', 'one_wgan_gp'
PRETRAIN_COARSE_NETWORK: False
GAN_LOSS_ALPHA: 0.001  # dcgan: 0.0008, wgan: 0.0005, onegan: 0.001
WGAN_GP_LAMBDA: 10
COARSE_L1_ALPHA: 2
L1_LOSS_ALPHA: 2
AE_LOSS_ALPHA: 2
GAN_WITH_MASK: False
DISCOUNTED_MASK: True
RANDOM_SEED: False
PADDING: 'SAME'

##training
NUM_GPUS: 1
GPU_ID: -1  # -1 indicate select any available one, otherwise select gpu ID, e.g. [0,1,3]
TRAIN_SPE: 750
MAX_ITERS: 100000
VIZ_MAX_OUT: 10
GRADS_SUMMARY: False
GRADIENT_CLIP: False
GRADIENT_CLIP_VALUE: 0.1
VAL_PSTEPS: 50

##data
DATA_FLIST:
  celebahq: [
    'training_data/train_shuffled.flist',
    'training_data/validation_static_view.flist'
  ]

STATIC_VIEW_SIZE: 30
IMG_SHAPES: [224, 224, 3]
HEIGHT: 224     
WIDTH: 224   
MAX_DELTA_HEIGHT: 0
MAX_DELTA_WIDTH: 0
BATCH_SIZE: 2
VERTICAL_MARGIN: 0
HORIZONTAL_MARGIN: 0

##loss
AE_LOSS: True
L1_LOSS: True
GLOBAL_DCGAN_LOSS_ALPHA: 1.
GLOBAL_WGAN_LOSS_ALPHA: 1.

##loss legacy
LOAD_VGG_MODEL: False
VGG_MODEL_FILE: data/model_zoo/vgg16.npz
FEATURE_LOSS: False
GRAMS_LOSS: False
TV_LOSS: False
TV_LOSS_ALPHA: 0.
FEATURE_LOSS_ALPHA: 0.01
GRAMS_LOSS_ALPHA: 50
SPATIAL_DISCOUNTING_GAMMA: 0.9

**Comment by JiahuiYu** [comment]
@123liluky You will need to set `stateful=True` in your following code. For the reasons, please checkout TensorFlow release document about this API.
```py
mask = tf.py_func(npmask, [bbox, height, width, config.MAX_DELTA_HEIGHT, config.MAX_DELTA_WIDTH], tf.float32, stateful=False)
```
